src/plugins/data_table/data_table.js

```diff
@@ -540,11 +573,13 @@ var LocalDataTable = (function() {
     _onDraw : function() {
       this.trigger("draw", arguments);
       this._renderGrandTotalsRow();
+      this._renderHeaderGroup();
     },
 
     _onColumnVisibilityChange: function(summary) {
       this.dataTable.find(".dataTables_empty").attr("colspan", summary.visible.length);
       this._renderGrandTotalsRow();
+      this._renderHeaderGroup();
```

Comment:
please add a test to cover this case (esp since it tripped us up).

ie confirm your header, then programmatically set visible(false) on one of the columns, and check that the DOM has hidden that whole header and sub columns.